src/Common/tests/Collections/IListTest.cs

```diff
@@ -0,0 +1,1692 @@
+                    list.Add(tempItems[i]);
+                }
+                for (int i = 0; i < 16; i++)
+                {
+                    if ((i & 1) == 0)
+                    {
+                        list.RemoveAt(i/2);
+                    }
+                    else
+                    {
+                        items[i/2] = tempItems[i];
+                    }
+                }
+
+                list.RemoveAt(list.Count - 1);
+                for (int i = 0; i < 16; i++)
+                {
+                    list.Add(tempItems[16 + i]);
+                }
+
+                Array.Copy(tempItems, 16, items, 7, 16);
+
+                CollectionAssert.Equal(items, list);
+            }
+        }
+
+        [Fact]
+        public void AddAndRemoveAll()
+        {
+            var list = GetList(null);
+            var tempItems = GenerateItems(16);
+            list.AddRange(tempItems);
+            foreach (object item in tempItems)
+            {
+                list.Remove(item);
+            }
+
+            object[] items = GenerateItems(16);
+            list.AddRange(items);
+            CollectionAssert.Equal(items, list);
+        }
+
+        [Fact]
+        public void AddWithInvalidTypes()
+        {
+            if (IsGenericCompatibility)
+            {
+                var items = GenerateItems(32);
+                var list = GetList(items);
+                foreach (var invalid in GetInvalidValues())
+                {
+                    Type[] expectedExceptions;
+                    if (ExpectedIsFixedSize)
+                    {
+                        expectedExceptions = new[]
+                        {
+                            typeof (ArgumentException),
+                            typeof (NotSupportedException)
+                        };
+                    }
+                    else
+                    {
+                        expectedExceptions = new[]
+                        {
+                            typeof (ArgumentException)
+                        };
+                    }
+                    object invalid1 = invalid;
+                    AssertThrows(
+                        expectedExceptions,
+                        () => list.Add(invalid1));
+                    CollectionAssert.Equal(items, list);
+                }
+            }
+        }
+
+        [Fact]
+        public void ClearWithReadOnlyOrFixedSizeThrows()
+        {
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+            {
+                var items = GenerateItems(32);
+                var list = GetList(items);
+
+                Assert.Throws<NotSupportedException>(() => list.Clear());
+                CollectionAssert.Equal(items, list);
+            }
+        }
+
+        [Fact]
+        public void ClearOnEmptyList()
+        {
+            if (!ExpectedIsFixedSize && !ExpectedIsReadOnly)
+            {
+                var list = GetList(null);
+                list.Clear();
```

Comment:
I don't quite understand what's being tested, just that there's no exception?

If so, I'd add a comment after that, and then assert the Count is 0.